### PR TITLE
Transforms fix type errors

### PR DESCRIPTION
**Pull request by rbuckton**
Various fixes for transforms.

Cc: @vladima

### CODE REVIEW

File: Jakefile.js [hunk 1]
```diff
@@ -1031,3 +1159,85 @@ task("lint-server", ["build-rules"], function() {
         lintWatchFile(lintTargets[i]);
     }
 });
+
+function ProgressBar(title) {
```
Comment (sandersn) [inline Jakefile.js]: Shouldn't this be in its own file? And I thought we already used an existing progress bar. Why write our own?

Comment (rbuckton) [inline Jakefile.js]: That one is part of a mocha reporter. This Is used by a task that runs mocha using the TAP reporter so there is no progress. It's just something small I threw together so you know something is still happening.

Comment (sandersn) [inline Jakefile.js]: Could we use [progress](https://www.npmjs.com/package/progress) instead?

Comment (rbuckton) [inline Jakefile.js]: That library doesn't work well when running node in Powershell.

**Pull request merged**